### PR TITLE
[go] Disable Remote JS debugger on SDK 49+

### PR DESCRIPTION
**Pull request by gabrieldonadel**
# Why

As part of https://github.com/facebook/react-native/pull/36754, we should remove the ability to use remote JS debugging from the DevMenu. This change is motivated by the fact that generally speaking, this feature does not work with the new architecture and most of the popular modules these days. 

Follow up of https://github.com/expo/expo/pull/22010
Closes ENG-8088

# How

This PR removes the `Remote JS debugger` option from Expo Go menu when using SDK 49 or above 

# Test Plan

Run Expo Go locally 
 
<table>
    <tr><th>iOS</th><th>Android</th></tr>
    <tr>
    <td>
        
   </td>
   <td>
     
    </td>
</tr> 
</table> 
  

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Review by Kudo — COMMENTED** [review]
i'm afraid that the sdkVersion from manifest could also be `UNVERSIONED`. could you help to double check whether this pr well handling this case?

to test unversioned NCL:
```sh
$ cd apps/native-component-list/
$ EXPO_SDK_VERSION=UNVERSIONED npx expo start
```

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,11 @@ Package-specific changes not released in any SDK will be added here just before
 
 ### 🐛 Bug fixes
 
+### ⚠️ Notices
+
+- **`expo-image-picker`**
```
[the remark this answers is not in the document]
Comment (gabrieldonadel) [reply CHANGELOG.md]: Opss my bad, I've removed the `expo-image-picker` part

**Comment by gabrieldonadel** [comment]
> i'm afraid that the sdkVersion from manifest could also be `UNVERSIONED`. could you help to double check whether this pr well handling this case?
> 
> to test unversioned NCL:
> 
> ```shell
> $ cd apps/native-component-list/
> $ EXPO_SDK_VERSION=UNVERSIONED npx expo start
> ```

I believe it's fine as we hide the `Debug Remote JS` when the SDK version is  `UNVERSIONED` 

image

image

**Comment by Kudo** [comment]
> I believe it's fine as we hide the `Debug Remote JS` when the SDK version is `UNVERSIONED`

thanks for the confirmation!

**Review by Kudo — APPROVED** [review]

**Comment by joshhogg** [comment]
Is there a proposed working alternative to removing the JS Debugger from Expo? This is a commonly used feature of the react native ecosystem, and many developers use it to help with debugging network request, inspecting layouts and using the redux developer plugin, alongside

**Comment by gabrieldonadel** [comment]
Hi @joshhogg, Remote JS debugging was basically broken if you use Turbo Modules, Fabric, or sync native module calls and therefore incompatible with most of the popular modules these days. 

People tend to assume that if something is part of the dev tools UI, it is going to work reliably. However, when it comes to remote debugging using JSC that's increasingly unlikely to be the case. Having it continue to exist as an option has created some confusion.

We've submitted a PR to update React native docs (https://github.com/facebook/react-native-website/pull/3702) to make this removal a bit clearer. If you're using Hermes there are many alternatives to how you can debug your app, one of them is just pressing `j` in your terminal running `npx expo start`, in case you're still using JSC you can use Direct debugging with Safari (https://reactnative.dev/docs/next/debugging#safari-developer-tools)

**Pull request merged**